src/parser/DASHTree.cpp

```diff
@@ -123,6 +123,8 @@ static unsigned int ParseSegmentTemplate(const char** attr,
   {
     if (!tpl.media.empty() && tpl.media[0] == '/')
```

Comment:
I would move the <!tpl.media.empty()> up into the parent IF, as this should be a common check for all conditions.  Additionally, it would be consistent with <tpl.initialization> below, plus the potential performance benefit.